@here/harp-mapview/lib/copyrights/UrlCopyrightProvider.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AreaCopyrightInfo, CopyrightCoverageProvider } from "./CopyrightProvider";
+
+/**
+ * Copyright provider which retrieves copyright coverage information from provided URL.
+ */
+export class UrlCopyrightProvider extends CopyrightCoverageProvider {
```

Comment:
To avoid a lot of subclasses we could use the same design principle as for OmvDataSource (i.e. have a separate CopyrightDataProvider). Might even make sense to talk to the Hype team and see if they could extent their HypeDataProvider or provide a HypeCopyrightProvider